sdk/projects/azure-projects/azure/projects/_resource.py

```diff
@@ -0,0 +1,632 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+from __future__ import annotations
+
+import inspect
+from itertools import product
+from collections import defaultdict
+import os
+from typing import (
+    Literal,
+    Mapping,
+    MutableMapping,
+    Tuple,
+    Optional,
+    Type,
+    Union,
+    Dict,
+    List,
+    Any,
+    Sequence,
+    Protocol,
+    AsyncContextManager,
+    ContextManager,
+    overload,
+    runtime_checkable,
+    TYPE_CHECKING,
+)
+from typing_extensions import Self, NamedTuple, TypedDict, TypeVar
+
+from dotenv import dotenv_values
+
+from azure.core.settings import _unset
+
+from .resources._identifiers import ResourceIdentifiers
+from ._utils import (
+    run_coroutine_sync,
+    resolve_properties,
+    find_last_resource_match,
+    find_resource_group,
+)
+from ._setting import StoredPrioritizedSetting
+from ._bicep.expressions import (
+    Output,
+    Parameter,
+    ResourceSymbol,
+    ResourceGroup as RGSymbol,
+)
+from ._bicep.utils import clean_name
+
+if TYPE_CHECKING:
+    from azure.core.credentials import SupportsTokenInfo
+    from azure.core.credentials_async import AsyncSupportsTokenInfo
+
+    from .resources.resourcegroup import ResourceGroup
+    from .resources._extension import RoleAssignment
+    from ._component import AzureInfrastructure
+
+
+def _load_dev_environment(name: Optional[str] = None, label: Optional[str] = None) -> Dict[str, Optional[str]]:
+    azd_dir = os.path.join(os.getcwd(), ".azure")
```

Comment:
Nit: I'm a pathlib person :p

Reply:
This whole function needs to go actually - I've replaced it with an alternative call into azd to load the env.